### ISSUE TITLE
aeroacoustics：plot_diricitivity.py can't run properly

### ISSUE THREAD

**Issue by 1codecopyer** [OpenFAST/python-toolbox#78]
Hello, everyone.
When I used Git to clone python- toolbox from OpenFAST repository and run the plot _ directivity.py , an official example under subpackage called aeroacoustics. Although it can run, I am puzzled by the running results, as shown in the figure:
![image](https://github.com/OpenFAST/python-toolbox/assets/115353218/20ec6349-b05f-49e2-a748-36802ceec327)
Viewing input file IEA_ LB_ RWT AeroAcoustics_ 1. out, where the parameter Number of observers and the number of data columns obs are 2
![image](https://github.com/OpenFAST/python-toolbox/assets/115353218/0ac57fee-062b-4f42-9140-b441c1082ab8)

The condition num_ obs<3 is true, run the 'print ("Error: Need at least 3 observers to generate contour.")' code (i.e. the tricontourf function has not been executed)

![image](https://github.com/OpenFAST/python-toolbox/assets/115353218/e9d846e7-1840-48be-ba8d-3f5ef4bf5f38)

The important parameter num_ obs is affected by AA_ 1. Shape [1] . I debug found AA_ 1. shape [1]==3, AA_ 1. shape [1] -1<3
![image](https://github.com/OpenFAST/python-toolbox/assets/115353218/9ce8c0df-5786-41f0-922b-88bf47e8052c)

Later, I tried other input files such as IEA_ LB_ RWT AeroAcoustics_ i. out (i=1 2 3 4), found that Number of observers: 2, all satisfy num_ obs<3

**Summary**：
This code can only run under conditions that meet num_obs < 3 .The code has an output result of print ("Error: Need at least 3 observers to generate source."), which does not allow for the normal implementation of the source plot of SPL for each location function. Is plot_ directivity.py needs to modify , If so, how can I modify it?

I modify the parameters in the code (such as num_obs=AA_1. shape [1], or modify if num_obs<2) to ensure that the else condition is true, with the aim of running the code after else. However, the code still reported an error, and the error prompt is:
![image](https://github.com/OpenFAST/python-toolbox/assets/115353218/928ad4a5-f0f0-48b0-a476-063bc9aa4179)

**Comment by ebranlard** [comment]
Thank you for reporting this. The script provided in the aeroacoustic package are mostly there to provide examples and get the user started. They have not been updated for a while and I can indeed expect them not to be the most robust. 

Ideally, this package should be restructured in a way similar to the other packages, see for instance the package `postpro` [here](https://github.com/OpenFAST/python-toolbox/tree/main/pyFAST/postpro) with library code put in python files at the root of the directory (the folder `aeroacoustics`), and with subfolders  `tests`, and `examples`. 

The current scripts provided are mostly "examples". These examples are standalone and do not use library code. 

I'll do a small reorganization of the folder now.

**Comment by ebranlard** [comment]
I've now done some small reorganizing of the folder (in the [dev branch ](https://github.com/OpenFAST/python-toolbox/tree/dev/pyFAST/aeroacoustics )of this repository). The problems you observed will still be there (with the scripts now in the `examples` folder).  

If you want feel, free to submit pull requests to progressively add library code to the file `aa_tools.py` or introduce your changes to make the examples scripts more robust. Ideally, the library code should be robust functions that can be reused, and the examples would simply make use of these functions. 

I will recommend to do very small pull requests at first, to give us the opportunity to give feedback along the way.

**Comment by 1codecopyer** [comment]
Thank you very much for your reply. By the way, apart from the running issues with the sample script, other modules such as case_ generation and fast.farm also have issues with being unable to run. Overall, there is still some room for improvement in the sample code of Python toolbox. Python toolbox changes the traditional way cmd runs to run OpenFAST and TurboSim. I still sincerely thank you, the R&D personnel, for your efforts.
![image](https://github.com/OpenFAST/python-toolbox/assets/115353218/98fd1d1c-5cc2-4de4-b675-dc597b5875d5)

**Comment by ebranlard** [comment]
Thank you for using the python-toolbox. We have limited ressources to work on these tools, so issues are expected. We appreciate community contributions as github issues or preferably pull requests. 

For the issue you mention above, I've now commited a fix. You might be using a python version more recent than what our test suite currently tests for. The issue comes from the fact that `collections.Iterable` was deprecated and moved to `collections.abc.Iterable`.  Hopefully it works now. Feel free to post separate issues as you find new bugs, or even pull requests if you know how to fix them.   Thanks!
